### ISSUE TITLE
Feature detection does not respect return statements

### ISSUE THREAD

**Issue by rvock** [amilajack/eslint-plugin-compat#523]
I like to use early return statements in functions, if a feature is unsupported. This leads to incorrect warnings:

```js
function version1() {
	if (window.fetch === undefined) {
		return;
	}

	// triggers a warning
	window.fetch();
}

function version2() {
	if (window.fetch !== undefined) {
		// triggers no warning
		window.fetch();
	}
}
```

I prefer `version1` because the remaining code within the function does not need to be indented and thus is easier to read/understand.

**Comment by tujoworker** [comment]
Here is a [PR](https://github.com/amilajack/eslint-plugin-compat/pull/676) to allow detection inside if statements. Hope it will be reviews by the maintainers.
